src/components/AuthorizationButton.tsx

```diff
@@ -0,0 +1,51 @@
+import { Button } from '@mui/material'
+import RedditIcon from '@mui/icons-material/Reddit'
+import { redirectToAutorizationPage, request } from '../services/OAUTH2'
+import { callbackURL, redditEndpoints, requestTypes } from '../constants'
+import { ApplicationCredentials } from '../types'
+import { APPNAME, API, SECRET } from '../keys.json'
+import { useAsyncEffect } from '../hooks/useAsyncEffect'
+import { getQueryParameter } from '../services/utils'
+import { useDispatch } from 'react-redux'
+import { setUserAction } from '../store/slices/userSlice'
+import { useNavigate } from 'react-router-dom'
+import { setBearerAction } from '../store/slices/bearerSlice'
+
+export const AuthorizationButton = () => {
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+
+  const authorizationData: ApplicationCredentials = {
+    appName: APPNAME,
+    clientId: API,
+    secret: SECRET,
+  }
+
+  useAsyncEffect(async () => {
+    if (getQueryParameter('code') !== null) {
```

Comment:
Functions where you have mixed dispatching actions and requests data is better to move into redux thunks

Reply:
fixed

Reply:
fixed